src/main/scala/com/fulcrumgenomics/testing/VariantContextSetBuilder.scala

```diff
@@ -0,0 +1,163 @@
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+
+package com.fulcrumgenomics.testing
+
+import java.nio.file.Files
+
+import dagr.commons.CommonsDef._
+import htsjdk.samtools.SAMSequenceDictionary
+import htsjdk.variant.variantcontext._
+import htsjdk.variant.variantcontext.writer.{Options, VariantContextWriterBuilder}
+import htsjdk.variant.vcf.{VCFFileReader, VCFHeader, VCFHeaderLine}
+
+import scala.collection.JavaConversions._
+import scala.collection.mutable.ListBuffer
+
+object VariantContextSetBuilder {
+  def apply(sampleName: String): VariantContextSetBuilder = {
+    new VariantContextSetBuilder(List(sampleName))
+  }
+}
+
+/**
+  * A class to make artificial variant contexts and VCFs just a little bit easier to generate.
+  *
+  * This builder uses the default sequence dictionary from [SamRecordSetBuilder] by default.
+  */
+class VariantContextSetBuilder(sampleNames: Seq[String] = List("Sample")) extends Iterable[VariantContext] {
+
+  if (sampleNames.isEmpty) throw new IllegalArgumentException("At least one samle name must be given")
+
+  private var _header = {
+    val h = new VCFHeader(Set.empty[VCFHeaderLine], sampleNames.toList)
+    h.setSequenceDictionary(new SamRecordSetBuilder().header.getSequenceDictionary)
+    h
+  }
+
+  private var variants = new ListBuffer[VariantContext]()
+
+  def header: VCFHeader = this._header
+
+  /** Sets the VCF header for this builder.  This should be set before adding variants. */
+  def setHeader(header: VCFHeader): this.type = {
+    this._header = header
+    this
+  }
+
+  /** Sets the sequence dictionary for this builder.  This should be set before adding variants. */
+  def setSequenceDictionary(dict: SAMSequenceDictionary): this.type = {
+    this._header.setSequenceDictionary(dict)
+    this
+  }
+
+  /** Adds the header line to the header for this builder.  This should be set before adding variants. */
+  def addMetaDataLine(headerLine: VCFHeaderLine): this.type = {
+    this._header.addMetaDataLine(headerLine)
+    this
+  }
+
+  /** Add a variant. */
+  def addVariant(refIdx: Int = 0,
```

Comment:
This could use some documentation since it's not clear to me how this works if your builder has > 1 sample without digging into the code, since you pass in `genotypeAlleles` and `sampleName`.